Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -3602,11 +3602,13 @@ template <class TreeBuilder> typename TreeBuilder::ImportSpecifier Parser<LexerT
 template <typename LexerType>
 template <class TreeBuilder> typename TreeBuilder::ImportAssertionList Parser<LexerType>::parseImportAssertions(TreeBuilder& context)
 {
+    HashSet<UniquedStringImpl*> keys;
```

Comment:
Can't we just use a boolean here?

Reply:
We cannot do that since we need to detect duplicate keys.